proxy.go

```diff
@@ -47,15 +47,39 @@ var sandboxID string
 
 var proxyLog = logrus.New()
 
+// This function is meant to run in a go routine since it will send ping
+// commands every second. It behaves as a heartbeat to maintain a proper
+// communication state with the Yamux server in the agent.
+func heartBeat(session *yamux.Session) {
+	if session == nil {
+		return
+	}
+
+	for {
+		if session.IsClosed() {
+			break
+		}
+
+		session.Ping()
+
+		// 1 Hz heartbeat
+		time.Sleep(time.Second)
+	}
+}
+
 func serve(servConn io.ReadWriteCloser, proto, addr string, results chan error) (net.Listener, error) {
 	sessionConfig := yamux.DefaultConfig()
 	// Disable keepAlive since we don't know how much time a container can be paused
 	sessionConfig.EnableKeepAlive = false
+	sessionConfig.ConnectionWriteTimeout = time.Second
```

Comment:
Any particular reason you set this - the default is 10s in the Yamux configs I think.

Reply:
By default, the ping command will wait for 10s to realize a write failed, and I thought it would be better to reduce this to 1s as we're not killing the communication if such error now happens. If we keep this to 10s by default, we might end up with cases where no heartbeat will be sent during 11s.